### PR TITLE
Bug 1152758 - [Messages] Empty report panel when opening it for an SMS t...

### PR DESCRIPTION
**Pull request by julienw**
...hat has been sent/received from a SIM that's not in the device anymore r=schung

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ed8cdf00736fda39dc8d39734ae6f17e0b23ac2f

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=26118e2d68d510fcbbd73feecf8177b06d535d15

File: apps/sms/js/information.js [hunk 1]
```diff
@@ -149,7 +149,9 @@ function showSimInfo(element, iccId) {
   var info = [];
   var simId = Settings.getServiceIdByIccId(iccId);
```
Comment (steveck-chung) [inline apps/sms/js/information.js]: After some experiment, both simId and operator will unable to be fetched if the iccId could not match any sim in the slot and the detailString will always empty. Maybe we can fetch the icc first and early return if not exist?

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c1eef7672b18608c63919c0f9faaec58fad74b3c

File: apps/sms/locales/sms.en-US.properties [hunk 2]
```diff
@@ -289,6 +289,7 @@ message-draft-saved  = Message saved as draft
 # some DSDS related localizations
 sim-id-label = SIM {{id}}
 sim-detail = SIM {{id}}, {{detailString}}
+dsds-unknown-sim = unknown SIM
```
Comment (julienw) [inline apps/sms/locales/sms.en-US.properties]: I think it's better without a capital letter at the start

File: apps/sms/js/information.js [hunk 3]
```diff
@@ -146,29 +146,34 @@ function showSimInfo(element, iccId) {
...
-  var operator = Settings.getOperatorByIccId(iccId);
-  var number = iccManager.getIccById(iccId).iccInfo.msisdn;
-  var data = {};
-  var l10nId;
 
-  info = [operator, number].filter(function(value) {
-    return value;
-  });
+  if (simId === null) {
```
Comment (julienw) [inline apps/sms/js/information.js]: I realized that we can never have operator or number if the SIM is unknown, so instead I changed the code like this. Tell me what you think !

Comment (steveck-chung) [inline apps/sms/js/information.js]: I agree we should show the unknown sim string directly so lgtm!

File: apps/sms/js/information.js [hunk 4]
```diff
@@ -146,29 +146,34 @@ function showSimInfo(element, iccId) {
...
+    navigator.mozL10n.setAttributes(
+      simInfoElement,
+      'dsds-unknown-sim'
```
Comment (steveck-chung) [inline apps/sms/js/information.js]: nit: I understand this only happend while dsds case that we need to show the sim info, but maybe we can simply use `unknown-sim` as id in general?

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#IK4hOJssQO-NtOw1kRGCSw

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#a23Fr8jFSUauzBO87J8d2A

The pull request failed to pass integration tests. It could not be landed, please try again.

**Pull request merged**